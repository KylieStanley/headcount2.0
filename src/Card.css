.card {
  align-items: center;
  border: 1px solid black;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  margin: 5px;
  padding: 8px;
  width: 250px;
}

ul {
  list-style-type: none;
  padding: 0;
}

h2 {
  text-align: center;
}

.greater-than {
  font-weight: bold;
  color: blue;
}

.less-than {

}