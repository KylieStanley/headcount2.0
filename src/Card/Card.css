.selected { 
  box-shadow: 0 0 13px 3px #000000ad;
}


.card {
  align-items: center;
  border: 1px solid black;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 340px;
  margin: 5px;
  padding: 20px 30px;
  width: 250px;
}

ul {
  list-style-type: none;
  padding: 0;
}

h2 {
  margin: 0;
  text-align: center;
}

.greater-than {
  font-weight: bold;
  color: blue;
}

.less-than {

}