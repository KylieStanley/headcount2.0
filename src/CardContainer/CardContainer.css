.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-items: center;
  border-radius: 20px;
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 5px;
  padding: 8px;
  text-align: center;
  width: 250px;
}