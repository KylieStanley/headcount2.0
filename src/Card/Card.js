import React from 'react';
import './Card.css';

const Card = ({ selectCard, district, selectedCards}) => {
  let selectedStyle = 'blue'

  return (
    <div className={`card ${selectedStyle}`} onClick={selectCard.bind(null, district)}>
      <h2>{district.location}</h2>
      <ul>
      {
        Object.keys(district.stats).map((item, i) => {
          let style = district.stats[item] > .5 ? 'greater-than' : 'less-than';
          return <li className={style} key={i}>{item}: {district.stats[item]}</li>
        })
      }
      </ul>
    </div>
  );
}

export default Card;
