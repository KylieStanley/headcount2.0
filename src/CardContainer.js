import React from 'react';
import Card from './Card';
import './CardContainer.css'

const CardContainer = (props) => {
  
    return (
      <div className="card-container">
        {
          props.data.map((district, i) => {
            return <Card district={district} key={i} />
          })
        }
      </div>
    );
  }

export default CardContainer;
