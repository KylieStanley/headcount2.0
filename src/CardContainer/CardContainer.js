import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css'

const CardContainer = ({ data, selectCard, selectedCards }) => {

  const allCards = data.map((district, i) => {
                  return <Card district={district} key={i} selectCard={selectCard} selectedCards={selectedCards}/>
                })

  return (
    <div className="card-container">
        { allCards }
    </div>
  );
}

export default CardContainer;
