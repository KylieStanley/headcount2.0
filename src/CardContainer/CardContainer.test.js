import React from 'react';
import Data from '../data/kindergartners_in_full_day_program.js';
import Card from '../Card/Card.js';
import CardContainer from './CardContainer.js';
import {shallow, mount} from 'enzyme';


describe('CardContainer', () => {
  it('should match the snapshot with data passed in correctly', () => {
    const wrapper = shallow( <CardContainer 
      data={[{location: "COLORADO",
              stats: {2005: .5}
            }]}
      selectCard={jest.fn()}
      selectedCards={[{location: "COLORADO SPRINGS",
                       stats: {2005: .4}
                      }] }
      compareDistrictAverages={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should return an array of cards', () => {
    const wrapper = mount( <CardContainer 
      data={[{location: "COLORADO",
              stats: {2004: .5}
            },{location: "COLORADO",
              stats: {2005: .5}
            }]}
      selectCard={jest.fn()}
      selectedCards={[{location: "COLORADO SPRINGS",
                       stats: {2005: .4}
                      }] }
      compareDistrictAverages={jest.fn()}
    />);

    expect(wrapper.find(Card).length).toBe(3)
  })
});