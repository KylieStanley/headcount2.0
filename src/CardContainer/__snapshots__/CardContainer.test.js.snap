// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    compareDistrictAverages={[Function]}
    data={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2005": 0.5,
              },
            },
          ]
    }
    selectCard={[Function]}
    selectedCards={
        Array [
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2005": 0.4,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="compare"
>
          <Card
                    district={
                              Object {
                                        "location": "COLORADO SPRINGS",
                                        "stats": Object {
                                          "2005": 0.4,
                                        },
                                      }
                    }
                    selectCard={[Function]}
          />
</div>,
        <div
          className="all-cards"
>
          <Card
                    district={
                              Object {
                                        "location": "COLORADO",
                                        "stats": Object {
                                          "2005": 0.5,
                                        },
                                      }
                    }
                    selectCard={[Function]}
          />
</div>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              district={
                            Object {
                                          "location": "COLORADO SPRINGS",
                                          "stats": Object {
                                            "2005": 0.4,
                                          },
                                        }
              }
              selectCard={[Function]}
/>,
          ],
          "className": "compare",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "district": Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2005": 0.4,
                },
              },
              "selectCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              district={
                            Object {
                                          "location": "COLORADO",
                                          "stats": Object {
                                            "2005": 0.5,
                                          },
                                        }
              }
              selectCard={[Function]}
/>,
          ],
          "className": "all-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "district": Object {
                "location": "COLORADO",
                "stats": Object {
                  "2005": 0.5,
                },
              },
              "selectCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="compare"
>
            <Card
                        district={
                                    Object {
                                                "location": "COLORADO SPRINGS",
                                                "stats": Object {
                                                  "2005": 0.4,
                                                },
                                              }
                        }
                        selectCard={[Function]}
            />
</div>,
          <div
            className="all-cards"
>
            <Card
                        district={
                                    Object {
                                                "location": "COLORADO",
                                                "stats": Object {
                                                  "2005": 0.5,
                                                },
                                              }
                        }
                        selectCard={[Function]}
            />
</div>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                district={
                                Object {
                                                "location": "COLORADO SPRINGS",
                                                "stats": Object {
                                                  "2005": 0.4,
                                                },
                                              }
                }
                selectCard={[Function]}
/>,
            ],
            "className": "compare",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "district": Object {
                  "location": "COLORADO SPRINGS",
                  "stats": Object {
                    "2005": 0.4,
                  },
                },
                "selectCard": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                district={
                                Object {
                                                "location": "COLORADO",
                                                "stats": Object {
                                                  "2005": 0.5,
                                                },
                                              }
                }
                selectCard={[Function]}
/>,
            ],
            "className": "all-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "district": Object {
                  "location": "COLORADO",
                  "stats": Object {
                    "2005": 0.5,
                  },
                },
                "selectCard": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
